hyperon_das/cache.py

```diff
@@ -135,7 +135,125 @@ def __next__(self):
                 if assignment_failed:
                     continue
                 assignment.freeze()
-
             lazy_query_answer.append(QueryAnswer(self._replace_target_handles(answer), assignment))
         self.buffered_answer = ListIterator(lazy_query_answer)
         return self.buffered_answer.__next__()
+
+
+class TraverseLinksIterator(QueryAnswerIterator):
+    def __init__(self, source: List[Tuple[Dict[str, Any], List[Dict[str, Any]]]], **kwargs) -> None:
+        super().__init__(source)
+        if not self.is_empty():
+            self.iterator = iter(source)
+            self.current_value = source[0][1]  # link
```

Comment:
This seems wrong. Firstly, this line is equal to line 156 below.
In addition to this, you are now checking the filters, as we discussed. The link you are putting in `current_value` is not guaranteed to be valid (i.e. to pass the filters)